### ISSUE TITLE
Evaluation results of fine-tuned DepthAntyhingV2 on Hypersim seem less than ideal

### ISSUE THREAD

**Issue by Namisntimpot** [DepthAnything/Depth-Anything-V2#261]
First of all, I would like to thank you for open-sourcing such an excellent project.

I tried testing your provided DepthAnythingV2 parameters fine-tuned on the Hypersim dataset on the Hypersim test set, and the results are as follows:
|MAE(L1) (in meter)|RMSE|AbsRel|$d_1$|
|:-:|:-:|:-:|:-:|
|1.26|1.55|0.21|0.64|

The results seem less than ideal. Does this meet your expectations? 

Table 4 of the DepthAnythingV2 paper shows good metric depth results obtained by fine-tuning on NYUv2 and KITTI. According to some other issues, those results are obtained by using ZoeDepth, which is provided in DepthAnythingV1 codebase. Do you have the results of using ZoeDepth to fine-tune DepthAnythingV2 on hypersim, or the results of using the scripts provided in DepthAnythingV2 codebase (i.e. retrain a DPT from scratch) to fine-tune on NYUv2, for a better comparison?

![Image](https://github.com/user-attachments/assets/5e1f4d6e-7e0c-4a3a-9bb7-be3b0c3c6982)

**Comment by VidhiRambhia** [comment]
Hi, I have experienced something similar as well

**Comment by zhangjy328** [comment]
I have also discovered a similar problem. During my actual testing, I found that the distance estimation in a narrow and close space is extremely inaccurate. I suspect that the model predicts the relative distance and converts it into the true value by using the proportion of the change in the maximum distance scale. So, does it mean that the prediction result is accurate only when the maximum distance in the image is 20 meters? And the greater the deviation, the more inaccurate the prediction result will be.

**Comment by sselgrad23** [comment]
I'm getting the same problem with the KITTI dataset. I altered the evaluation code to add these lines in eval_depth, after the `assert pred.shape==target.shape` line: 
```
    epsilon = 1e-6
    pred = torch.clamp(pred, min=epsilon)
    target = torch.clamp(target, min=epsilon)
```

I got these results somehow:

```
[2025-03-18 02:02:51,776][    INFO] load ckpt from ./checkpoints/depth_anything_v2_vitl.pth.
mae: 9.887: 100%|██████████| 652/652 [03:23<00:00,  3.21it/s]
[2025-03-18 02:06:17,947][    INFO] ==========================================================================================
[2025-03-18 02:06:17,947][    INFO]       d1,       d2,       d3,  abs_rel,   sq_rel,     rmse, rmse_log,    log10,    silog,      mae
[2025-03-18 02:06:17,947][    INFO]    0.125,    0.279,    0.461,    1.138,   13.420,   10.059,    0.782,    0.302,    0.608,    9.342
[2025-03-18 02:06:17,947][    INFO] ==========================================================================================

```

**Comment by SarahAnberlin** [comment]
> I'm getting the same problem with the KITTI dataset. I altered the evaluation code to add these lines in eval_depth, after the `assert pred.shape==target.shape` line:
> 
> ```
>     epsilon = 1e-6
>     pred = torch.clamp(pred, min=epsilon)
>     target = torch.clamp(target, min=epsilon)
> ```
> 
> I got these results somehow:
> 
> ```
> [2025-03-18 02:02:51,776][    INFO] load ckpt from ./checkpoints/depth_anything_v2_vitl.pth.
> mae: 9.887: 100%|██████████| 652/652 [03:23<00:00,  3.21it/s]
> [2025-03-18 02:06:17,947][    INFO] ==========================================================================================
> [2025-03-18 02:06:17,947][    INFO]       d1,       d2,       d3,  abs_rel,   sq_rel,     rmse, rmse_log,    log10,    silog,      mae
> [2025-03-18 02:06:17,947][    INFO]    0.125,    0.279,    0.461,    1.138,   13.420,   10.059,    0.782,    0.302,    0.608,    9.342
> [2025-03-18 02:06:17,947][    INFO] ==========================================================================================
> ```

Facing the same issue. And also, I found the predicted depth is shown as an inversed manner w.r.t. the gt depth. I've tried to also inverse the gt or even add scale shift adjustment, but it just doesn't do anything good.
